Client/Internal/MeasurementMapper.cs

```diff
@@ -35,18 +35,30 @@ internal PointData ToPoint<TM>(TM measurement, WritePrecision precision)
 
             var measurementType = measurement.GetType();
             CacheMeasurementClass(measurementType);
-            
+
             var measurementAttribute = (Measurement) measurementType.GetCustomAttribute(typeof(Measurement));
-            if (measurementAttribute == null)
+            var measurementColumn = CACHE[measurementType.Name].SingleOrDefault(p => p.Column.IsMeasurement);
+
+            if ((measurementAttribute == null) ^ (measurementColumn == null) == false)
```

Comment:
Please change expression to, for better maintenance and understanding by other users:

```suggestion
            if (((measurementAttribute == null) ^ (measurementColumn == null)) == false)
```